### PR TITLE
Upgrade brcm syncd to buster.

### PR DESCRIPTION
**Pull request by vmittal-msft**
**- Why I did it**
To upgrade brcm syncd to buster

**- How I did it**
1. Updated BCM SAI using kernel version 4.19.0-12 and debian 10 to support buster.
2. Updated syncd docker from stretch to buster in sonic-buildimage

**- How to verify it**
1.  Verified change on S6000 Dell system. Ensured docker is running synd buster.
2. After upgrade, ensured all BGP peers and ip interfaces are up.
3. Ping to BGP neighbors is working fine.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

Upgrade syncd from stretch to buster

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by jleveque — CHANGES_REQUESTED** [review]
As part of this PR, please also switch to using Python 3 to run supervisord_dependent_startup and supervisor-proc-exit-listener. In the file platform/broadcom/docker-syncd-brcm/supervisord.conf, please change the line

```python
command=python2 -m supervisord_dependent_startup
```

to

```python
command=python3 -m supervisord_dependent_startup
```

And change the line

```python
command=python2 /usr/bin/supervisor-proc-exit-listener --container-name syncd
```

to

```python
command=/usr/bin/supervisor-proc-exit-listener --container-name syncd
```

**Comment by smaheshm** [comment]
retest vs please

File: platform/broadcom/sai.mk [hunk 1]
```diff
@@ -1,8 +1,8 @@
-BRCM_SAI = libsaibcm_4.2.1.5-4_amd64.deb
-$(BRCM_SAI)_URL = "https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.2/master/libsaibcm_4.2.1.5-4_amd64.deb?sv=2015-04-05&sr=b&sig=dHfQf95FxgLBlnrGJwv6pIkAYAc6mOGbz9vvUMfxFvg%3D&se=2034-07-28T05%3A53%3A21Z&sp=r"
-BRCM_SAI_DEV = libsaibcm-dev_4.2.1.5-4_amd64.deb
+BRCM_SAI = libsaibcm_4.2.1.5_amd64-buster.deb
+$(BRCM_SAI)_URL = "https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.2/master/libsaibcm_4.2.1.5_amd64-buster.deb?sv=2019-12-12&st=2020-11-24T21%3A45%3A37Z&se=2021-11-25T21%3A45%3A00Z&sr=b&sp=r&sig=6we05VowbABC4VhxUOQlZdnfohRgqpOMUxYdL9O4sX0%3D"
+BRCM_SAI_DEV = libsaibcm-dev_4.2.1.5_amd64-buster.deb
```
Comment (lguohan) [inline platform/broadcom/sai.mk:3]: can you put buster in the url path/folder?

Comment (vmittal-msft) [reply platform/broadcom/sai.mk:3]: Sure. Will update.

**Comment by vmittal-msft** [comment]
> As part of this PR, please also switch to using Python 3 to run supervisord_dependent_startup and supervisor-proc-exit-listener. In the file platform/broadcom/docker-syncd-brcm/supervisord.conf, please change the line
> 
> ```python
> command=python2 -m supervisord_dependent_startup
> ```
> 
> to
> 
> ```python
> command=python3 -m supervisord_dependent_startup
> ```
> 
> And change the line
> 
> ```python
> command=python2 /usr/bin/supervisor-proc-exit-listener --container-name syncd
> ```
> 
> to
> 
> ```python
> command=/usr/bin/supervisor-proc-exit-listener --container-name syncd
> ```

Done.

**Comment by lguohan** [comment]
please add proper description of the pr.

File: platform/broadcom/sai.mk (hunk 1 above)
Comment (vmittal-msft) [reply platform/broadcom/sai.mk]: Done

**Comment by smaheshm** [comment]
can you also fill the questions in the description.

- Why I did it
- How I did it
- How to verify it
- Description for the changelog

File: platform/broadcom/docker-ptf-brcm.mk [hunk 2]
```diff
@@ -5,4 +5,4 @@ $(DOCKER_PTF_BRCM)_PATH = $(DOCKERS_PATH)/docker-ptf-saithrift
 $(DOCKER_PTF_BRCM)_DEPENDS += $(PYTHON_SAITHRIFT)
 $(DOCKER_PTF_BRCM)_LOAD_DOCKERS += $(DOCKER_PTF)
 SONIC_DOCKER_IMAGES += $(DOCKER_PTF_BRCM)
-SONIC_STRETCH_DOCKERS += $(DOCKER_PTF_BRCM)
+SONIC_BUSTER_DOCKERS += $(DOCKER_PTF_BRCM)
```
Comment (jleveque) [inline platform/broadcom/docker-ptf-brcm.mk:8]: I don't think it is necessary to append to `SONIC_BUSTER_DOCKERS`. I don't think we reference this variable anywhere. All Dockers are considered Buster by default, unless appended to `SONIC_STRETCH_DOCKERS`. @lguohan to confirm.

Comment (vmittal-msft) [reply platform/broadcom/docker-ptf-brcm.mk:8]: Sure. Will wait for @lguohan to confirm.

**Comment by vmittal-msft** [comment]
retest broadcom please

File: platform/broadcom/docker-ptf-brcm.mk (hunk 2 above)
Comment (lguohan) [reply platform/broadcom/docker-ptf-brcm.mk:8]: PTF stay on STRETCH.

File: platform/broadcom/platform-modules-delta.mk [hunk 3]
```diff
@@ -40,4 +40,4 @@ DELTA_AGC032_PLATFORM_MODULE = platform-modules-agc032_$(DELTA_AGC032_PLATFORM_M
 $(DELTA_AGC032_PLATFORM_MODULE)_PLATFORM = x86_64-delta_agc032-r0
 $(eval $(call add_extra_package,$(DELTA_AG9032V1_PLATFORM_MODULE),$(DELTA_AGC032_PLATFORM_MODULE)))
 
-SONIC_STRETCH_DEBS += $(DELTA_AG9032V1_PLATFORM_MODULE)
+SONIC_BUSTER_DEBS += $(DELTA_AG9032V1_PLATFORM_MODULE)
```
Comment (lguohan) [inline platform/broadcom/platform-modules-delta.mk:43]: we can do that in a different pr.

File: platform/broadcom/docker-ptf-brcm.mk (hunk 2 above)
Comment (lguohan) [reply platform/broadcom/docker-ptf-brcm.mk:8]: @jleveque , yes you are right.

Comment (vmittal-msft) [reply platform/broadcom/docker-ptf-brcm.mk]: Done.

File: platform/broadcom/platform-modules-delta.mk (hunk 3 above)
Comment (vmittal-msft) [reply platform/broadcom/platform-modules-delta.mk]: Done. @lguohan Shall i open a different PR for check in these changes ? or is there already one i can use ?

**Review by lguohan — APPROVED** [review]

**Comment by lguohan** [comment]
@jleveque , can you approve?

File: platform/broadcom/docker-saiserver-brcm.mk [hunk 4]
```diff
@@ -4,9 +4,9 @@ DOCKER_SAISERVER_BRCM = docker-saiserver-brcm.gz
 $(DOCKER_SAISERVER_BRCM)_PATH = $(PLATFORM_PATH)/docker-saiserver-brcm
 $(DOCKER_SAISERVER_BRCM)_DEPENDS += $(SAISERVER)
 $(DOCKER_SAISERVER_BRCM)_FILES += $(DSSERVE) $(BCMCMD)
-$(DOCKER_SAISERVER_BRCM)_LOAD_DOCKERS += $(DOCKER_CONFIG_ENGINE_STRETCH)
+$(DOCKER_SAISERVER_BRCM)_LOAD_DOCKERS += $(DOCKER_CONFIG_ENGINE_BUSTER)
 SONIC_DOCKER_IMAGES += $(DOCKER_SAISERVER_BRCM)
-SONIC_STRETCH_DOCKERS += $(DOCKER_SAISERVER_BRCM)
+SONIC_BUSTER_DOCKERS += $(DOCKER_SAISERVER_BRCM)
```
Comment (jleveque) [inline platform/broadcom/docker-saiserver-brcm.mk:9]: As discussed in a previous comment, we do not reference a `SONIC_BUSTER_DOCKERS` variable. This line is unnecessary, correct, @lguohan?

File: platform/broadcom/docker-syncd-brcm-rpc.mk [hunk 5]
```diff
@@ -12,7 +12,7 @@ endif
 $(DOCKER_SYNCD_BRCM_RPC)_FILES += $(DSSERVE) $(BCMCMD) $(SUPERVISOR_PROC_EXIT_LISTENER_SCRIPT)
 $(DOCKER_SYNCD_BRCM_RPC)_LOAD_DOCKERS += $(DOCKER_SYNCD_BASE)
 SONIC_DOCKER_IMAGES += $(DOCKER_SYNCD_BRCM_RPC)
-SONIC_STRETCH_DOCKERS += $(DOCKER_SYNCD_BRCM_RPC)
+SONIC_BUSTER_DOCKERS += $(DOCKER_SYNCD_BRCM_RPC)
```
Comment (jleveque) [inline platform/broadcom/docker-syncd-brcm-rpc.mk:15]: As discussed in a previous comment, we do not reference a `SONIC_BUSTER_DOCKERS` variable. This line is unnecessary, correct, @lguohan?

**Review by jleveque — CHANGES_REQUESTED** [review]

File: platform/broadcom/docker-saiserver-brcm.mk (hunk 4 above)
Comment (vmittal-msft) [reply platform/broadcom/docker-saiserver-brcm.mk]: Done, reverted the change.

File: platform/broadcom/docker-syncd-brcm-rpc.mk (hunk 5 above)
Comment (vmittal-msft) [reply platform/broadcom/docker-syncd-brcm-rpc.mk]: Done. Reverted the change.

File: platform/broadcom/docker-saiserver-brcm.mk [hunk 6]
```diff
@@ -4,7 +4,7 @@ DOCKER_SAISERVER_BRCM = docker-saiserver-brcm.gz
 $(DOCKER_SAISERVER_BRCM)_PATH = $(PLATFORM_PATH)/docker-saiserver-brcm
 $(DOCKER_SAISERVER_BRCM)_DEPENDS += $(SAISERVER)
 $(DOCKER_SAISERVER_BRCM)_FILES += $(DSSERVE) $(BCMCMD)
-$(DOCKER_SAISERVER_BRCM)_LOAD_DOCKERS += $(DOCKER_CONFIG_ENGINE_STRETCH)
+$(DOCKER_SAISERVER_BRCM)_LOAD_DOCKERS += $(DOCKER_CONFIG_ENGINE_BUSTER)
 SONIC_DOCKER_IMAGES += $(DOCKER_SAISERVER_BRCM)
 SONIC_STRETCH_DOCKERS += $(DOCKER_SAISERVER_BRCM)
```
Comment (jleveque) [inline platform/broadcom/docker-saiserver-brcm.mk:9]: Shouldn't you still delete this line? This is now a Buster Docker, not Stretch, correct?

File: platform/broadcom/docker-syncd-brcm-rpc.mk (hunk 5 above)
Comment (jleveque) [reply platform/broadcom/docker-syncd-brcm-rpc.mk]: I think you should still delete the line, just not add a `SONIC_BUSTER_DOCKERS` line.

File: platform/broadcom/docker-saiserver-brcm.mk (hunk 4 above)
Comment (jleveque) [reply platform/broadcom/docker-saiserver-brcm.mk]: I think you should still delete the line, just not add a `SONIC_BUSTER_DOCKERS` line.

File: platform/broadcom/docker-saiserver-brcm.mk (hunk 6 above)
Comment (vmittal-msft) [reply platform/broadcom/docker-saiserver-brcm.mk:9]: Just confirming that "$(DOCKER_SAISERVER_BRCM)_LOAD_DOCKERS += $(DOCKER_CONFIG_ENGINE_STRETCH)" need to be removed too ?

Comment (vmittal-msft) [reply platform/broadcom/docker-saiserver-brcm.mk:9]: and not adding "$(DOCKER_SAISERVER_BRCM)_LOAD_DOCKERS += $(DOCKER_CONFIG_ENGINE_BUSTER)" ?

Comment (jleveque) [reply platform/broadcom/docker-saiserver-brcm.mk]: I assumed you are now basing this container off Buster. If so, then I believe this section should read:

```
DOCKER_SAISERVER_BRCM = docker-saiserver-brcm.gz
$(DOCKER_SAISERVER_BRCM)_PATH = $(PLATFORM_PATH)/docker-saiserver-brcm
$(DOCKER_SAISERVER_BRCM)_DEPENDS += $(SAISERVER)
$(DOCKER_SAISERVER_BRCM)_FILES += $(DSSERVE) $(BCMCMD)
$(DOCKER_SAISERVER_BRCM)_LOAD_DOCKERS += $(DOCKER_CONFIG_ENGINE_BUSTER)
SONIC_DOCKER_IMAGES += $(DOCKER_SAISERVER_BRCM)
```

**Comment by jleveque** [comment]
@vmittal-msft: FYI, for future PRs, I suggest only using force-pushes when necessary (i.e., after a rebase), because we lose track of the commit history.

**Review by jleveque — APPROVED** [review]

**Comment by vmittal-msft** [comment]
retest vs please

**Comment by vmittal-msft** [comment]
retest vsimage please

**Comment by gechiang** [comment]
retest vsimage please

**Comment by gechiang** [comment]
retest vs please

**Comment by gechiang** [comment]
retest vsimage please
